pennylane/transforms/tape_expand.py

```diff
@@ -180,3 +187,81 @@ def expand_fn(tape, _depth=depth, **kwargs):
     stop_at=not_tape | is_measurement | (~is_trainable) | has_grad_method,
     docstring=_expand_invalid_trainable_doc,
 )
+
+
+@contextlib.contextmanager
+def _custom_decomp_context(custom_decomps):
+    """A context manager for applying custom decompositions. Used by the
+    create_custom_decomp_expand_fn method."""
+
+    # Creates an individual context
+    @contextlib.contextmanager
+    def _custom_decomposition(obj, fn):
+        # Covers the case where the user passes a string to indicate the Operator
+        if isinstance(obj, str):
+            obj = getattr(qml, obj)
+
+        original_decomp_method = obj.decompose
+
+        # This is the method that will override the operations .decompose method
+        def new_decomp_method(self):
+            with NonQueuingTape() as tape:
+                if self.num_params == 0:
+                    return fn(self.wires)
+                return fn(*self.parameters, self.wires)
+            return tape
+
+        try:
+            # Actually set the new decomp function; note that we set
+            # expand rather than `decomposition` proper.
```

Comment:
Wait, why? Aren't we replacing `decompose`?

Reply:
maybe an outdated comment?

Reply:
^ Yup, forgot to update this one, my bad.